sklearnex/cluster/dbscan.py

```diff
@@ -15,4 +15,167 @@
 # limitations under the License.
 # ===============================================================================
 
-from daal4py.sklearn.cluster import DBSCAN
+import numbers
+from abc import ABC
+
+import numpy as np
+from scipy import sparse as sp
+from sklearn.cluster import DBSCAN as sklearn_DBSCAN
+from sklearn.utils.validation import _check_sample_weight
+
+from daal4py.sklearn._utils import sklearn_check_version
+from onedal.cluster import DBSCAN as onedal_DBSCAN
+
+from .._device_offload import dispatch, wrap_output_data
+
+if sklearn_check_version("1.1") and not sklearn_check_version("1.2"):
+    from sklearn.utils import check_scalar
+
+
+class BaseDBSCAN(ABC):
+    def _save_attributes(self):
+        assert hasattr(self, "_onedal_estimator")
+
+        self.labels_ = self._onedal_estimator.labels_
+        self.core_sample_indices_ = self._onedal_estimator.core_sample_indices_
+        self.components_ = self._onedal_estimator.components_
+        self.n_features_in_ = self._onedal_estimator.n_features_in_
+
+
+class DBSCAN(sklearn_DBSCAN, BaseDBSCAN):
+    __doc__ = sklearn_DBSCAN.__doc__
+
+    if sklearn_check_version("1.2"):
+        _parameter_constraints: dict = {**sklearn_DBSCAN._parameter_constraints}
+
+    def __init__(
+        self,
+        eps=0.5,
+        *,
+        min_samples=5,
+        metric="euclidean",
+        metric_params=None,
+        algorithm="auto",
+        leaf_size=30,
+        p=None,
+        n_jobs=None,
+    ):
+        super(DBSCAN, self).__init__(
+            eps=eps,
+            min_samples=min_samples,
+            metric=metric,
+            metric_params=metric_params,
+            algorithm=algorithm,
+            leaf_size=leaf_size,
+            p=p,
+            n_jobs=n_jobs,
+        )
+        self.eps = eps
+        self.min_samples = min_samples
+        self.metric = metric
+        self.metric_params = metric_params
+        self.algorithm = algorithm
+        self.leaf_size = leaf_size
+        self.p = p
+        self.n_jobs = n_jobs
+
+    def _onedal_fit(self, X, y, sample_weight=None, queue=None):
+        onedal_params = {
+            "eps": self.eps,
+            "min_samples": self.min_samples,
+            "metric": self.metric,
+            "metric_params": self.metric_params,
+            "algorithm": self.algorithm,
+            "leaf_size": self.leaf_size,
+            "p": self.p,
+            "n_jobs": self.n_jobs,
+        }
+        self._onedal_estimator = onedal_DBSCAN(**onedal_params)
+
+        self._onedal_estimator.fit(X, y=y, sample_weight=sample_weight, queue=queue)
+        self._save_attributes()
+
+    def _onedal_cpu_supported(self, method_name, *data):

```

Comment:
Patching tracking is required for non-preview functionality, for example: [kNN](https://github.com/intel/scikit-learn-intelex/blob/master/sklearnex/neighbors/common.py#L143).

Reply:
Also, `cpu` and `gpu -supported` methods are equal and can be merged even simpler than in kNN

Reply:
Please be kind provide this kind of comments before.
This is non-necessary comment that will patched in the next PR.